main/src/cgeo/geocaching/ui/ImagesList.java

```diff
@@ -153,17 +170,65 @@ public void onClick(final View arg0) {
             activity.registerForContextMenu(imageView);
             imageView.setImageDrawable(image);
             imageView.setScaleType(ImageView.ScaleType.CENTER_CROP);
-            imageView.setLayoutParams(new LinearLayout.LayoutParams(bounds.width(), bounds.height()));
+            imageView.setLayoutParams(new RelativeLayout.LayoutParams(bounds.width(), bounds.height()));
 
             view.findViewById(R.id.progress_bar).setVisibility(View.GONE);
 
             imageView.setId(image.hashCode());
+            addGeoOverlay(imageViewLayout, img);
             images.put(imageView.getId(), img);
 
             view.invalidate();
         }
     }
 
+    private void addGeoOverlay(final RelativeLayout imageViewLayout, final Image img) {
+        try {
+            // Extract metadata.
+            final File file = LocalStorage.getStorageFile(geocode, img.getUrl(), true, true);
```

Comment:
Last parameter should be `false`. Also, `getStorageFile` cannot return `null`, I'll add an annotation there.